Mage.Sets/src/mage/cards/t/ThreefoldSignal.java

```diff
@@ -0,0 +1,81 @@
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.costs.mana.GenericManaCost;
+import mage.abilities.effects.common.continuous.EachSpellYouCastHasReplicateEffect;
+import mage.abilities.effects.keyword.ScryEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.cards.ModalDoubleFacesCardHalf;
+import mage.cards.SplitCardHalf;
+import mage.constants.CardType;
+import mage.constants.Zone;
+import mage.filter.FilterSpell;
+import mage.filter.predicate.Predicate;
+import mage.game.Game;
+
+import java.util.UUID;
+
+/**
+ * @author Alex-Vasile
+ */
+public class ThreefoldSignal extends CardImpl {
+
+    private static final FilterSpell filter = new FilterSpell("spell you cast that's exactly three colors");
+    static {
+        filter.add(ThreeColorPredicate.instance);
+    }
+
+    public ThreefoldSignal(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ARTIFACT}, "{3}");
+
+        // When Threefold Signal enters the battlefield, scry 3.
+        this.addAbility(new EntersBattlefieldTriggeredAbility(new ScryEffect(3)));
+
+        // Each spell you cast that’s exactly three colors has replicate {3}.
+        //  (When you cast it, copy it for each time you paid its replicate cost.
+        //   You may choose new targets for the copies.
+        //   A copy of a permanent spell becomes a token.)
+        this.addAbility(new SimpleStaticAbility(new EachSpellYouCastHasReplicateEffect(filter, new GenericManaCost(3))));
+    }
+
+    private ThreefoldSignal(final ThreefoldSignal card) {
+        super(card);
+    }
+
+    @Override
+    public ThreefoldSignal copy() {
+        return new ThreefoldSignal(this);
+    }
+}
+
+/**
+ * Based on MultiColorPredicate
+ */
+enum ThreeColorPredicate implements Predicate<MageObject> {
+    instance;
+
+    @Override
+    public boolean apply(MageObject input, Game game) {
+        // 708.3. Each split card that consists of two halves with different colored mana symbols in their mana costs
+        // is a multicolored card while it's not a spell on the stack. While it's a spell on the stack, it's only the
+        // color or colors of the half or halves being cast. #
+        if (input instanceof SplitCardHalf
+                && game.getState().getZone(input.getId()) != Zone.STACK) {
+            return 3 == ((SplitCardHalf) input).getMainCard().getColor(game).getColorCount();
+        } else if (input instanceof ModalDoubleFacesCardHalf
+                && (game.getState().getZone(input.getId()) != Zone.STACK && game.getState().getZone(input.getId()) != Zone.BATTLEFIELD)) {
+            // While a double-faced card isn’t on the stack or battlefield, consider only the characteristics of its front face.
+            return 3 == ((ModalDoubleFacesCardHalf) input).getMainCard().getColor(game).getColorCount();
+        } else {
+            return 3 == input.getColor(game).getColorCount();
```

Comment:
does this work correctly with split cards and MDFCs? it should only care about the part that's being cast

Reply:
Idk about MDFCs (see comment above), as for split cards, we shall see when I add tests.

Reply:
did you get around to writing these tests?

Reply:
Not yet!

I plan to get to them this weekend.

In the meantime let me know what else you find

Reply:
Hmm. It is not interacting correctly with a fused spell.
If I try casting [[Beck // Call]] [Beck // Call](https://scryfall.com/card/dgm/123/beck-call) fused and pay the replicate cost once, the stack looks like this:
<img width="228" alt="Screen Shot 2022-09-16 at 10 34 25 PM" src="https://user-images.githubusercontent.com/48962821/190837407-9054ed19-0b0a-4870-8c98-65028544a2f6.png">

One of those two must be wrong since I expect them to be identical.

But it's also fundamentally wrong. It seems to copy Beck only once but Call seems to be copied twice. I.e. when I resolve **only** the copy I will end up with 8 birds and 8 ETB triggers on the stack, even though Call should only be making 4 birds.

Reply:
These tests work properly now. But I still need to check if the visual bug (the original cast and copy looking different) is still there.

Reply:
I checked copying with Reverberate and there were no visual bugs, either in the MTGO-style or image renderers

Reply:
> I checked copying with Reverberate and there were no visual bugs, either in the MTGO-style or image renderers

Must have been fixed by your PR